dev/local-environment-dynamic/configurations/cardano/entrypoint.sh

```diff
@@ -0,0 +1,1163 @@
+        rm -f "$DELEG_CERT" "$DELEG_TX_FINAL"
+        continue
+    fi
+
+    DELEG_TX_SIGNED="/data/${NODE_LOG_NAME}_deleg_tx.signed"
+
+     if ! cardano-cli latest transaction sign \
+        --tx-body-file "$DELEG_TX_FINAL" \
+        --signing-key-file "$NODE_PAYMENT_SKEY" \
+        --signing-key-file "$NODE_STAKE_SKEY" \
+        --testnet-magic 42 \
+        --out-file "$DELEG_TX_SIGNED"; then
+        echo "[DEBUG] ERROR: Failed to sign delegation transaction for $NODE_LOG_NAME."
+        echo "[DEBUG] Attempted to use keys:"
+        echo "[DEBUG]   Payment SKey: $NODE_PAYMENT_SKEY"
+        echo "[DEBUG]   Stake SKey:   $NODE_STAKE_SKEY"
+        rm -f "$DELEG_CERT" "$DELEG_TX_FINAL"
+        continue
+    fi
+
+    DELEG_TX_ID=$(cardano-cli latest transaction txid --tx-file "$DELEG_TX_SIGNED")
+    if [ -z "$DELEG_TX_ID" ]; then
+        echo "[DEBUG] CRITICAL ERROR: Could not get TxID for delegation tx for $NODE_LOG_NAME. Cannot save final UTXO."
+        rm -f "$DELEG_CERT" "$DELEG_TX_FINAL" "$DELEG_TX_SIGNED"
+        continue
+    fi
+
+    # 15. Submit Delegation Transaction
+    echo "[LOG] Submitting delegation transaction for $NODE_LOG_NAME..."
+    SUBMITTED_DELEG=false
+    for attempt in {1..5}; do
+        if cardano-cli latest transaction submit --tx-file "$DELEG_TX_SIGNED" --testnet-magic 42; then
+            echo "[LOG] Delegation transaction submitted for $NODE_LOG_NAME."
+            SUBMITTED_DELEG=true
+            break
+        else
+            echo "[WARN] Attempt $attempt to submit delegation transaction for $NODE_LOG_NAME failed. Retrying in 5s..."
+            sleep 5
+        fi
+    done
+
+    rm -f "$DELEG_CERT" "$DELEG_TX_FINAL" "$DELEG_TX_SIGNED" # Clean up
+
+    if [ "$SUBMITTED_DELEG" = false ]; then
+        echo "[DEBUG] CRITICAL ERROR: Failed to submit delegation transaction for $NODE_LOG_NAME after multiple attempts."
+    else
+        FINAL_UTXO="${DELEG_TX_ID}#0"
+        echo "[LOG] Saving final UTXO for $NODE_LOG_NAME to /shared/registered-${i}.utxo: $FINAL_UTXO"
+        echo "$FINAL_UTXO" > "/shared/registered-${i}.utxo"
+
+        echo "[LOG] Waiting for delegation transaction for $NODE_LOG_NAME ($DELEG_TX_ID) to be confirmed on-chain..."
+        for attempt in {1..20}; do
+            # Check if the UTXO created by the delegation transaction exists.
+            if cardano-cli latest query utxo --tx-in "$DELEG_TX_ID#0" --testnet-magic 42 | /busybox grep -q "$DELEG_TX_ID"; then
+                echo "[LOG] Delegation transaction for $NODE_LOG_NAME confirmed."
+                break
+            fi
+            if [ "$attempt" -eq 20 ]; then
+                echo "[DEBUG] CRITICAL ERROR: Delegation transaction for $NODE_LOG_NAME was not confirmed on-chain after 100 seconds."
+            fi
+            sleep 5
+        done
+
+        echo "[LOG] Waiting for stake delegation for $NODE_LOG_NAME to become active (2 epochs)..."
+        sleep 250
```

Comment:
Please don't wait after each stake delegation. Waiting 2 epoch is implemented elsewhere.
I can see this query stake-address-info below, but it is not worth time lost here.